developer.md

```diff
@@ -0,0 +1,114 @@
+# Developer notes
+
+## Cloning the project
+
+```bash
+git clone --recursive https://github.com/RediSearch/RediSearch.git
+```
+
+## Installing prerequisites
+
+## Build dependencies
+To build and test RediSearch you need to install several packages, depending on the underlying OS. The following OSes are supported and tested in our CI:
+
+Ubuntu 18.04
+Ubuntu 20.04
+Ubuntu 22.04
+Ubuntu 24.04
+Debian linux 11
+Debian linux 12
+Rocky linux 8
+Rocky linux 9
+Amazon linux 2
+Amazon linux 2023
+Mariner 2.0
+MacOS
+Alpine linux 3
+
+For installing the prerequisites you can take the following approaches:
+1. Install the dependencies manually - you can find the dependencies in each OS script under the [`.install`](.install) directory.
+2. Use our CI installation scripts to install the dependencies - you can find the scripts under the [`.install`](.install) directory.
+
+    ```bash
+    cd ./install
+    ./install_script.sh sudo
+    ./install_boost.sh 1.83.0
+    ```
+    Note that this will install various packages on your system using the native package manager (sudo is not required in a Docker environment). 
+
+## Redis
+You will not be able to run and test your code without Redis, since you need to load the module. You can build it from source and install it as described in [redis GitHub page](https://github.com/redis/redis).
+
+#### RedisJSON
+Some of our behavioral tests require RedisJSON to be present. Our testing framework will clone and build RedisJSON for you.
+
+
+
+## Building from source
+To build RediSearch from source, you need to run the following commands:
+
+```bash
+make build
```

Comment:
Do we want to mention other flags? (COORD, CLANG, ...)

Reply:
Or mention `make help`?

Reply:
not for now